src/data/equiv/algebra.lean

```diff
@@ -190,6 +199,112 @@ protected def discrete_field [discrete_field β] : discrete_field α :=
 end instances
 end equiv
 
+-- Do this definition and two lemmas go somewhere else?
+def is_add_hom {α β : Type*} [has_add α] [has_add β] (f : α → β) : Prop :=
+∀ x y, f (x + y) = f x + f y
+
+lemma is_add_hom.id (α : Type*) [has_add α] : is_add_hom (id : α → α) := λ _ _, rfl
+
+lemma is_add_hom.comp [has_add α] [has_add β] [has_add γ] {f : α → β}
+  {g : β → γ} (hf : is_add_hom f) (hg : is_add_hom g) : is_add_hom (g ∘ f) :=
+λ x y, by show _ = g _ + g _; rw [←hg, ←hf]
+
+structure add_equiv (α β : Type*) [has_add α] [has_add β] extends α ≃ β :=
+(hom : is_add_hom to_fun)
+
+infix ` ≃+ `:50 := add_equiv
+
+namespace add_equiv
+
+variables [has_add α] [has_add β] [has_add γ]
+
+@[refl] def refl (α : Type) [has_add α] : α ≃+ α :=
+{ hom := λ _ _,rfl,
+..equiv.refl _}
+
+@[symm] def symm (h : α ≃+ β) : β ≃+ α :=
+{ hom := λ n₁ n₂, function.injective_of_left_inverse h.left_inv begin
+   rw h.hom, unfold equiv.symm, rw [h.right_inv, h.right_inv, h.right_inv], end
+  ..h.to_equiv.symm}
+
+@[trans] def trans (h1 : α ≃+ β) (h2 : β ≃+ γ) : (α ≃+ γ) := {
+  hom := is_add_hom.comp h1.hom h2.hom,
```

Comment:
Curly bracket shod be on first line after the statement.

Reply:
done